crates/mongodb-agent-common/src/query/execute_query_request.rs

```diff
@@ -39,8 +39,14 @@ pub async fn execute_query_request(
             let collection = database.collection(&collection_name);
             collect_from_cursor(collection.aggregate(pipeline, None).await?).await
         }
-        QueryTarget::NativeQuery { .. } => {
-            collect_from_cursor(database.aggregate(pipeline, None).await?).await
+        QueryTarget::NativeQuery { native_query, .. } => {
+            match native_query.clone().collection_name {
+                Some(coll_name) => {
+                    let collection = database.collection(&coll_name.to_string());
```

Comment:
I think you can simplify this

```suggestion
                    let collection = database.collection(coll_name);
```